x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -114,14 +125,16 @@ static boolean isDiskOnlyNoDecision(Decision decision) {
         return nos.size() == 1 && DiskThresholdDecider.NAME.equals(nos.get(0).label());
     }
 
-    static class AllocationState {
+    // todo: move this to top level class.
```

Comment:
I started on this, but decided that it will confuse the reviewers of this more than it helps to do this up front. I will move this class to a top-level class as well as move the associated tests to a new `AllocationStateTests` in a follow-up.